packages/esm-patient-registration-app/src/patient-registration/field/name/name-field.component.tsx

```diff
@@ -55,6 +55,36 @@ export const NameField = () => {
     }
   };
 
+  const firstNameField = (
+    <Input
+      id="givenName"
+      name="givenName"
+      labelText={t('givenNameLabelText', 'First Name')}
+      checkWarning={checkNumber}
+      required
+    />
+  );
+
+  const middleNameField = (
+    <Input
+      id="middleName"
+      name="middleName"
+      labelText={t('middleNameLabelText', 'Middle Name')}
+      light
+      checkWarning={checkNumber}
+    />
+  );
+
+  const familyNameField = (
+    <Input
+      id="familyName"
+      name="familyName"
+      labelText={t('familyNameLabelText', 'Family Name')}
+      checkWarning={checkNumber}
+      required
+    />
+  );
```

Comment:
Then this could get whittled down to something like:

```suggestion
{nameKnown &&
  (reverseFieldOrder
    ? ['familyName', 'middleName', 'givenName']
    : ['givenName', 'middleName', 'familyName']
  ).map((field) => {
    if (field === 'middleName' && !fieldConfigs.displayMiddleName) {
      return null;
    }

    return (
      <Input
        id={field}
        name={field}
        labelText={t(
          `${field}LabelText`,
          field === 'givenName' ? 'First Name' : field === 'familyName' ? 'Family Name' : 'Middle Name',
        )}
        checkWarning={checkNumber}
        light={field === 'middleName'}
        required={field !== 'middleName'}
      />
    );
  })}
```

Reply:
The t(`${field}LabelText`) won't be taken up by the translator for the right fields.
This also complicates it with multiple === as well.
I believe we should go ahead defining the fields separately only.